packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -271,11 +271,12 @@ const EMPTY_PARAMS: BindableAttrEntries = [];
 
 function QueryEditor({
   globalScope,
-  connectionParams,
+  connectionParams: rawConnectionParams,
   value: input,
   onChange: setInput,
 }: QueryEditorProps<RestConnectionParams, FetchQuery>) {
-  const baseUrl = connectionParams?.baseUrl;
+  const connectionParams = input.attributes.query.value.browser ? null : rawConnectionParams;
```

Comment:
Maybe we can store `input.attributes.query.value.browser` as a `isClientSideQuery` variable to reuse in these 2 lines?